### DISCUSSION TITLE
SOAP calls timing out

### DISCUSSION THREAD

**Discussion by elidrissidev** [OpenMage/magento-lts#1919] [Questions & Answers]
Alright, so something strange happened lately, I was running an openmage instance locally at `openmage.local` along side other things like phpMyAdmin at `phpmyadmin.local` for a few months now but in the last week it stopped working completely, if I visit any of them chrome just shows `ERR_NAME_NOT_RESOLVED`. I thought a recent system update I did messed something up (I'm using Manjaro KDE btw) so I went checking if all services were running (`httpd`, `php-fpm`) and sure enough all of them were. My last resort was to change the virtual host name from `.local` to `.localhost` and they all started working again. But now SOAP calls were timing out, I followed the request using `xdebug` to identify what's causing this timeout and found out it was the call to `checkdnsrr` in `Zend_Validate_Hostname`: 
https://github.com/OpenMage/magento-lts/blob/4e7a8e134ab62f5b5532bec50890e3abf3809b6a/app/code/core/Zend/Validate/Hostname.php#L2503-L2527

If I switch the host name to something like `.com` or `.org` it will work just fine.

Anyone has an idea on what's going on here? why did this stop working all of a sudden?

**Comment by addison74** [comment]
Let's check the file changes history here https://github.com/OpenMage/magento-lts/commits/1.9.4.x/app/code/core/Zend/Validate/Hostname.php.

The latest commit is here and it is related to checkDnsRecords function. Try to revert this commit and see what is happening

https://github.com/OpenMage/magento-lts/commit/0412ad6e020289cfe43054bacecfc329f7892f73#diff-51a775794b786e994252b5043afb525ca7e0162f64c53b6c6ca58399003a1bab

**Reply by elidrissidev** [reply]
Thanks for the reply, but the issue is related to `checkdnsrr` call itself, as far as I can see it was present even before that commit, for some reason it no longer works with my virtual hosts!
E.g. when calling `php -r "var_dump(checkdnsrr('<host>', 'A'));"` with the following hosts:
- `localhost` exists in my system and the command will return true immediately.
- `openmage.local` does not exist in my system and the command will return false immediately.
- `openmage.localhost` exists in my system and the command will take several seconds before returning false!


![Screenshot_20211225_121850](https://user-images.githubusercontent.com/67818913/147383860-02acb342-b6aa-4711-b89a-60dd68de2485.png)

And in a single SOAP request there are several calls to `checkdnsrr` so it ends taking enough to timeout.
